### ISSUE TITLE
wrong LabelSwitchDelegate output 

### ISSUE THREAD

**Issue by ADevelopStudio** [Cookiezby/LabelSwitch#16]
in your file LabelSwitch.swift yo have wrong order in
` func switchTaped(sender: Any) `

your one: 
```
 @objc
    func switchTaped(sender: Any) {
        UIView.animate(withDuration: 0.3) {
            switch self.curState {
            case .L:
                self.delegate?.switchChangToState(sender: self)
                self.curState = .R
            case .R:
                self.delegate?.switchChangToState(sender: self)
                self.curState = .L
            }
        }
    }
```

you need to call delegate after changing `curState`. Otherwise the 
` func switchChangToState(sender: LabelSwitch) `  gives you wrong output 

ideally, I'd suggest:
```
    func switchTaped(sender: Any) {
        UIView.animate(withDuration: 0.3, animations: {
            switch self.curState {
            case .L:
                self.curState = .R
            case .R:
                self.curState = .L
            }
        }) { (completed) in
            if completed {
                self.delegate?.switchChangToState(sender: self)
            }
        }
    }

```

**Comment by Cookiezby** [comment]
Thank you for your advice, I will check these weekend :D

**Comment by Cookiezby** [comment]
fix on 0.1.7
